test/integration_test.py

```diff
@@ -286,73 +287,17 @@ def managed_access_sources_by_catalog(self
         """
-        plugin = self.azul_client.repository_plugin(catalog)
-        source = plugin.resolve_source(source)
-        prefix = source.spec.prefix
-        partition_prefixes = list(prefix.partition_prefixes())
-        partition_prefix = self.random.choice(partition_prefixes)
-        effective_prefix = prefix.common + partition_prefix
-        fqids = self.azul_client.list_bundles(catalog, source, partition_prefix)
-        num_bundles = len(fqids)
-        partition = f'Partition {effective_prefix!r} of source {source.spec}'
-        if not config.deployment.is_sandbox_or_personal:
-            # For sources that use partitioning, 512 is the desired partition
-            # size. In practice, we observe the reindex succeeding with sizes
-            # >700 without the partition size becoming a limiting factor. From
-            # this we project 1024 as a reasonable upper bound to enforce.
-            upper = 1024
-            if effective_prefix:
-                lower = 512 // 16
-                if len(fqids) < lower:
-                    # If bundle UUIDs were uniformly distributed by prefix, we
-                    # could directly assert a minimum partition size, but since
-                    # they're not, a given partition may fail to exceed this
-                    # minimum even with an optimal choice of partition prefix
-                    # length.
-                    log.warning('With %i bundles, %s is too small', len(fqids), partition)
-                    counts = plugin.list_partitions(source)
-                    if counts is None:
-                        lower = 1
-                    else:
-                        # For plugins that support efficiently counting bundles
-                        # across all partitions simultaneously, we can check
-                        # whether the chosen partition is an outlier by
-                        # determining the *average* partition size.
-                        num_bundles = sum(counts.values()) / prefix.num_partitions
-            else:
-                # Sources too small to be split into more than one partition may
-                # have as few as one bundle in total
-                lower = 1
-        else:
-            # The sandbox and personal deployments typically don't use
-            # partitioning and the desired number of bundles per source is
-            # 16 +/- 50%, i.e. 8 to 24. However, no choice of common prefix can
-            # satisfy this range for snapshots with `n` bundles where `n < 8` or
-            # `24 < n <= 112`. The choice of upper bound here is somewhat
-            # arbitrary.
-            upper = 64
-            lower = 1
-
-        self.assertLessEqual(num_bundles, upper, partition + ' is too large')
-        self.assertGreaterEqual(num_bundles, lower, partition + ' is too small')
-
-        return source, partition_prefix, fqids
-
-    def _list_bundles(self,
```

Comment:
A comment in `_prepare_notifications()` that references `_list_bundles()` needs to be updated.